src/Qir/Tests/QIR-dynamic/qsharp/qir-test-assert.qs

```diff
@@ -1,15 +1,235 @@
+    operation AsrtMeasSPlusMinusTest() : Unit {
+        use qubit = Qubit();    // |0>
+        H(qubit);               // |+>
+        S(qubit);               // (|0> + i|1>) / SQRT(2)
+        AssertMeasurement(           [PauliY], [qubit], Zero,      "0: Call failed");
+        // 50% probability in other Pauli bases:
+        AssertMeasurementProbability([PauliZ], [qubit], Zero, 0.5, "1: Call failed", 1e-10);
+        AssertMeasurementProbability([PauliZ], [qubit],  One, 0.5, "2: Call failed", 1e-10);
+        AssertMeasurementProbability([PauliX], [qubit], Zero, 0.5, "3: Call failed", 1e-10);
+        AssertMeasurementProbability([PauliX], [qubit],  One, 0.5, "4: Call failed", 1e-10);
+
+        Adjoint S(qubit);   // Back to |+>
+        H(qubit);   // Back to |0>
+        X(qubit);   // |1>
+        H(qubit);   // |->
+        S(qubit);   // (|0> - i|1>) / SQRT(2)
+        AssertMeasurement(           [PauliY], [qubit], One,       "5: Call failed");
+        // 50% probability in other Pauli bases:
+        AssertMeasurementProbability([PauliZ], [qubit], Zero, 0.5, "6: Call failed", 1e-10);
+        AssertMeasurementProbability([PauliZ], [qubit],  One, 0.5, "7: Call failed", 1e-10);
+        AssertMeasurementProbability([PauliX], [qubit], Zero, 0.5, "8: Call failed", 1e-10);
+        AssertMeasurementProbability([PauliX], [qubit],  One, 0.5, "9: Call failed", 1e-10);
+    }
+
+
+    // Multiple qubits:
+
+    // Quantum Katas, Joint Measurement Workbook,
+    //  https://hub.gke2.mybinder.org/user/microsoft-quantumkatas-jv0pu1ej/notebooks/JointMeasurements/Workbook_JointMeasurements.ipynb#Task-1.-Single-qubit-measurement
+
+    //  Task 1. Single-qubit measurement
+    //  Task 2. Parity measurement
+    // |0>, |1>, |->
+    @EntryPoint()
+    operation AsrtMeas0011() : Unit {
+        use left = Qubit();
+        use right = Qubit();
+
+        AssertMeasurement(           [PauliZ, PauliZ], [left, right], Zero,      "0: Call failed");
+        AssertMeasurementProbability([PauliZ, PauliZ], [left, right], Zero, 1.0, "1: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliZ, PauliZ], [left, right],  One, 0.0, "2: Call failed", 1E-05);
+
+        H(left);        // |+>
+        H(right);       // |+>
+        AssertMeasurement(           [PauliX, PauliX], [left, right],  Zero,     "B: Call failed");
+        // 50% probability in other Pauli bases:
+        AssertMeasurementProbability([PauliZ, PauliZ], [left, right], Zero, 0.5, "C: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliZ, PauliZ], [left, right],  One, 0.5, "D: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliY, PauliY], [left, right], Zero, 0.5, "E: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliY, PauliY], [left, right],  One, 0.5, "F: Call failed", 1E-05);
+        
+        H(left);        // Back to |0>
+        H(right);       // Back to |0>
```

Comment:
It would be much clearer to use a `within`/`apply` block here to make sure that the calls to `H` are correctly undone before the next group of assertions:

```suggestion
        within {
            H(left);        // |+>
            H(right);       // |+>
        } apply {
            AssertMeasurement(           [PauliX, PauliX], [left, right],  Zero,     "B: Call failed");
            // 50% probability in other Pauli bases:
            AssertMeasurementProbability([PauliZ, PauliZ], [left, right], Zero, 0.5, "C: Call failed", 1E-05);
            AssertMeasurementProbability([PauliZ, PauliZ], [left, right],  One, 0.5, "D: Call failed", 1E-05);
            AssertMeasurementProbability([PauliY, PauliY], [left, right], Zero, 0.5, "E: Call failed", 1E-05);
            AssertMeasurementProbability([PauliY, PauliY], [left, right],  One, 0.5, "F: Call failed", 1E-05);
        }
```

Reply:
Thanks. Fixed.